lua/toggleterm.lua

```diff
@@ -321,9 +293,56 @@ function _G.___toggleterm_winbar_click(id)
     term:toggle()
   end
 end
+
+function M.get_last_open_term()
+  local terminals = terms.get_all()
+  ---@type Terminal
+  local target
+  -- count backwards from the end of the list
+  for i = #terminals, 1, -1 do
+    local term = terminals[i]
+    if term and ui.term_has_open_win(term) then
+      target = term
+      break
+    end
+  end
+  if not target then
+    utils.notify("Couldn't find a terminal to close", "error")
+    return
+  end
+  return target.id
+end
+
+M.toggle_strategies = {
+  -- use the first terminal in the terminal list
+  first = function()
+    if not ui.find_open_windows() then
+      -- Re-open the first terminal toggled
+      return terms.get_toggled_id()
+    end
+    return M.get_last_open_term()
+  end,
+
+  -- use a terminal corresponding to the current tabpage handle
+  by_tabpage = function()
+    if not ui.find_open_windows() then
+      return vim.api.nvim_get_current_tabpage()
```

Comment:
I'm not sure I see how this works? Are you envisioning only one terminal per tab? This won't align with future plans to allow creating multiple terminals per tab

Reply:
This essentially has exactly the same effect as `smart_toggle` had before, it's just that we're using the current tab's handle as the terminal ID rather than the ID of the first terminal found in the list of unhidden terminals. It doesn't restrict you to one terminal per tab or anything, it just defines the default behavior of `:ToggleTerm` without a count.